iceoryx2-ffi/ffi/src/api/message_type_details.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use core::ffi::c_char;
+
+use iceoryx2::service::static_config::message_type_details::*;
+
+use crate::iox2_type_variant_e;
+
+#[derive(Clone, Copy)]
+#[repr(C)]
+pub struct iox2_type_detail_t {
+    pub variant: iox2_type_variant_e,
+    pub type_name: [c_char; 256],
+    pub size: usize,
+    pub alignment: usize,
+}
```

Comment:
I agree that the version is important, that's why the `DynamicStorage` concept has an internal version that corresponds to the iceoryx2 version. It is untouchable from the outside.

The `StaticStorage` still requires such a version and I wanted to encode this into the underlying name so that you have something like `myprefix_static_storage_name_v0.1.2_mysuffix`. I find a version number at this level to error prone, the question is, what structs require a version number and which doesn't. Also it can be forgotten.

From the iceoryx2 point of view it is perfect when the underlying concepts takes care of the versioning and then you do not have to worry about this on this level.

Reply:
I created an issue #327

Reply:
I'm more concerned about recod&replay. This information then also needs to be stored somewhere when record&replay is implemented.

Reply:
You are right! But I would keep those two related problems separated.
 1. Detecting version mismatches of multiple iceoryx2 applications.
   * In my opinion perfectly solved by the current approach.
 2. Versioning recorded data.
   * Here we could again some kind of record & replay concept that takes care of exactly this automatically. Here we may require a trait for all recorded data that implements some versioning information.